libs/wlgen/wlgen/rta.py

```diff
@@ -472,6 +489,50 @@ def conf(self,
 
         Custom workloads
         ----------------
+        When 'kind' is 'custom', you can supply existing rt-app job descriptions
+        either directly or by supplying a file name pointing to the json.
+        To use a job description stored in a file, you can either set 'params'
+        to be a filename or you can have 'params' as a dictionary where
+        'custom_filename' holds the name . Note that the filename
+        is relative to the current directory.
+        To supply rt-app job description directly, assign the content to
+        'custom_config' inside 'params'. If params is a dictionary and
+        'custom_config' is present, 'custom_filename' will be ignored.
+
+        This allows you to reuse the push/run-test/pull functionality with
+        more complex application models not covered by the workload generator.
+
+        For example, to create an RTA wrapper for an existing rt-app task
+        description stored in a file whose name is held in a variable called
+        'filename':
+
+        generated_file_name = rtapp.conf(
+            kind='custom',
+            duration=-1,
+            params="{}".format(filename)
+            run_dir='/data/local/tmp' )
+
+
+        Or if it is convenient, params may be a dictionary:
+
+        generated_file_name = rtapp.conf(
+            kind='custom',
+            duration=-1,
+            params={
+	    	    'custom_filename' : "{}".format(filename)
+            },
+            run_dir='/data/local/tmp' )
+
+
+        Or to write the config into a variable directly:
+
+        generated_file_name = rtapp.conf(
+            kind='custom',
+            duration=-1,
+            params={
+                'custom_config'= { ...my description of the rt-app job... }
```

Comment:
Maybe you already considered that, isn't possible to avoid introducing a nested dictionary and just check the value of params?
```python
if ininstace(self.params['custom'], str):
    # we have the filename for a JSON configuraiton
elif ininstace(self.params['custom'], dict):
    # we have an embedded dictionary-cased configuration
else:
    raise ValueError("params can be only a filename or an embedded dictionary")
```